### DISCUSSION TITLE
Widlcard URL Pattern matching in an HTML attribute

### DISCUSSION THREAD

**Discussion by jmoppel** [unpoly/unpoly#477] [Q&A] [answered]
I'm having difficulties getting Unpoly to recognize a wildcard [URL pattern](https://unpoly.com/url-patterns) which I'd like to use to close an overlay after a successful new form submission. Here is the HTML link I'm having trouble with:

``` html
<a up-layer="new" up-accept-location="/cars/types/*uuid/" up-on-accepted="alert('New object created')" class="btn btn-link text-decoration-none ps-0" href="/cars/types/new/">New</a>
```

When I click on the link above, the new form overlay opens just fine. When I save the form, I see the POST happen, and a successful GET requested to the newly created item for the URL `/cars/types/34d11d42-1407-4417-b113-1e3516525d3f/`. The problem is the overlay remains open and does not close. I would expect this URL to match the pattern I specified in `up-accept-location="/cars/types/*uuid/"`, but the overlay remains open. Also, I'm not receiving an alert from `up-on-accepted`.

All the examples in the documentation I could find have examples for `$id`, which won't work in my case since UUIDs contain characters. Any help figuring out what I'm doing wrong would be really appreciated!

**Comment by adam12** [comment, marked as answer]
I'd suggest trying this URL format: `/cars/types/*/`. I think `$id` is a special case.

**Reply by triskweline** [reply]
Yes, `/cars/types/*/` will work.

If you also need to capture the UID you can use `/cars/types/:uid/`.

**Reply by jmoppel** [reply]
That works great, thank you both! For folks who might find this thread in the future, here is the entire solution:

``` html
<a up-layer="new" up-accept-location="/cars/types/:uuid/ -/cars/types/new/" up-on-accepted="alert('New object created')" class="btn btn-link text-decoration-none ps-0" href="/cars/types/new/">New</a>
```
I'd be happy to update the [URL pattern docs](https://unpoly.com/url-patterns) with some HTML and a couple more JS examples if that would be welcome? Either way, thanks again!

**Reply by szabeszg** [reply]
> I'd be happy to update the [URL pattern docs](https://unpoly.com/url-patterns) with some HTML and a couple more JS examples if that would be welcome?

I support it :)

**Reply by triskweline** [reply]
Contributions to the docs are always welcome.

All doc pages have an `EDIT THIS PAGE` that links to the source code from which the page was generated.
